### DISCUSSION TITLE
How to use VERCEL_URL Client Side in Preview Build

### DISCUSSION THREAD

**Discussion by rtnolan** [vercel/vercel#5219] [Help]
Hi there,

I can't quite figure out how to access the `VERCEL_URL` from client side in a Preview Build. I've tried setting an environment variable through the UI in Preview as `NEXT_PUBLIC_VERCEL_URL=${VERCEL_URL}` and have also tried creating a `.env.preview` file with the same assignment, but no joy. We deploy through Bitbucket and was wondering if this was possible? Any help would be appreciated.

Regards,
Tom

**Comment by philihp** [comment]
If you're coming from an API endpoint, another way to get the URL of the page is from the request, passing that to [`next-absolute-url`](https://www.npmjs.com/package/next-absolute-url)
